C7Engine/MapUnitExtensions.cs

```diff
@@ -67,15 +148,18 @@ public static bool fight(this MapUnit unit, MapUnit defender)
 
 	public static void bombard(this MapUnit unit, Tile tile)
 	{
-		MapUnit target = tile.findTopDefender(unit);
+		MapUnit target = tile.FindTopDefender(unit);
 		if ((unit.unitType.bombard == 0) || (target == MapUnit.NONE))
 			return; // Do nothing if we don't have a unit to attack. TODO: Attack city or tile improv if possible
 
 		var unitOriginalOrientation = unit.facingDirection;
 		unit.facingDirection = unit.location.directionTo(tile);
 
-		int defenderStrength = target.unitType.defense;
-		double attackerOdds = defenderStrength > 0 ? (double)unit.unitType.bombard / (unit.unitType.bombard + defenderStrength) : 1.0;
+		double bombardStrength  = unit  .AttackStrengthVersus (target, true, unit.facingDirection);
+		double defenderStrength = target.DefenseStrengthVersus(unit  , true, unit.facingDirection);
+		double attackerOdds = bombardStrength / (bombardStrength + defenderStrength);
```

Comment:
seems like we would also want BombardStrengthVersus and BombardDefenseVersus since those may have their own modifiers later on

Reply:
As it is now, the Attack/DefenseStrengthVersus functions do double duty, as they also compute strengths for bombarding depending on that boolean flag. Since then I have also added a DefensiveBombardStrengthVersus method. We could split these up into six methods, but actually I was thinking of doing the opposite and combining them all into one StrengthVersus method that would take an enum specifying the combat type (Attack, Defense, Bombard, BombardDefense, DefensiveBombard, or DefensiveBombardDefense). Otherwise I think we'll end up with a lot of code duplication since the strength bonuses will be similar in many cases.